lms/djangoapps/course_home_api/general/v1/views.py

```diff
@@ -0,0 +1,69 @@
+"""
+General view for the Course Home that contains information every page needs.
+"""
+
+
+from rest_framework.generics import RetrieveAPIView
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+
+from opaque_keys.edx.keys import CourseKey
+
+from lms.djangoapps.courseware.access import has_access
+from lms.djangoapps.courseware.tabs import get_course_tab_list
+from lms.djangoapps.course_api.api import course_detail
+from lms.djangoapps.course_home_api.general.v1.serializers import CourseHomeSerializer
+
+
+class CourseHomeView(RetrieveAPIView):
+    """
+    **Use Cases**
+
+        Request details for the Course Home MFE that every page needs.
+
+    **Example Requests**
+
+        GET api/course_home/v1/general/{course_key}
+
+    **Response Values**
+
+        Body consists of the following fields:
+
+        course_id: (str) The Course's id (Course Run key)
+        is_staff: (bool) Indicates if the user is staff
+        number: (str) The Course's number
+        org: (str) The Course's organization
+        tabs: List of Course Tabs to display. They are serialized as:
+            tab_id: (str) The tab's id
+            title: (str) The title of the tab to display
+            url: (str) The url to view the tab
+        title: (str) The Course's display title
+
+    **Returns**
+
+        * 200 on success with above fields.
+        * 403 if the user is not authenticated.
+        * 404 if the course is not available or cannot be seen.
+    """
+
+    permission_classes = (IsAuthenticated,)
```

Comment:
Hmm, we might want to leave this one open? Like, so we can at least render the shell of the page for a drive by user?

Reply:
I think right now, we automatically redirect a user anyway, but good point. It will eventually be used even by unauthenticated users. I'll update.